File: backend/schemas/warehouse.py
from __future__ import annotations

from typing import Optional
from pydantic import BaseModel, Field


class WarehouseBase(BaseModel):
    name: str
    address: str
    volume_m3: float
    latitude: float
    longitude: float
    iot_token: Optional[str] = None


class WarehouseCreate(WarehouseBase):
    pass


class WarehouseUpdate(WarehouseBase):
    status: Optional[str] = None


class WarehouseResponse(WarehouseBase):
    warehouse_id: str
    owner_id: Optional[int] = None
    status: str
    is_rented: bool = False


class WarehouseListItem(BaseModel):
    id: str
    name: str
    address: str
    volume: float
    lat: float
    lon: float
    gps: str
    status: str
    is_rented: bool

File: frontend/src/types/warehouse.ts
export interface WarehouseData {
  warehouse_id: string
  name: string
  address: string
  volume_m3: number
  latitude: number
  longitude: number
  status?: string
  is_rented?: boolean
}

export interface WarehouseCreate {
  name: string
  address: string
  volume_m3: number
  latitude: number
  longitude: number
}
